sklearn/impute/tests/test_impute.py

```diff
@@ -1531,6 +1531,18 @@ def test_iterative_imputer_keep_empty_features(initial_strategy):
     assert_allclose(X_imputed[:, 1], 0)
 
 
+@pytest.mark.parametrize("keep_empty_features", [True, False])
+def test_iterative_imputer_keep_empty_features_different_input_x(keep_empty_features):
+    """Check the behaviour of the iterative imputer with different keep_empty_features
+    and different input.
+    """
+    X = np.array([[np.nan, 0, 1], [2, np.nan, 3], [4, 5, np.nan]])
+
+    imputer = IterativeImputer(keep_empty_features=keep_empty_features)
+    X_imputed = imputer.fit_transform(X)
+    assert X_imputed.shape == (3, 3)
```

Comment:
I think a stronger test is to run both True and False in the same run and make sure the results are the same.

Reply:
Changed the test case to compare the outputs of when keep_empty_features is set to True and False are same or not.